client/components/FormattingBar/media/MediaTwitter.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import { InputGroup, Button, Intent, NonIdealState } from '@blueprintjs/core';
+import { isHttpsUri } from 'valid-url';
+
+import Icon from 'components/Icon/Icon';
+import { apiFetch } from 'client/utils/apiFetch';
+import { getIframeSrc, getEmbedType } from 'client/utils/editor';
+
+type Props = {
+	onInsert: (...args: any[]) => any;
+};
+
+const sampleUrl =
+	process.env.NODE_ENV === 'production'
+		? 'https://twitter.com/pubpub'
+		: 'https://twitter.com/kfutures';
+
+type State = any;
+
+class MediaTwitter extends Component<Props, State> {
+	constructor(props: Props) {
+		super(props);
+		this.state = {
+			isValid: false,
+			input: '',
+			embedUrl: '',
+			embedTitle: '',
+		};
+		this.handleInput = this.handleInput.bind(this);
+		this.handleInsert = this.handleInsert.bind(this);
+	}
+
+	handleInput(url) {
+		const input = url;
+		const isValid = isHttpsUri(input) && getEmbedType(input) === 'twitter';
+		this.setState(
+			{
+				input: input,
+				isValid: isValid,
+			},
+			() => {
+				if (!this.state.isValid) {
+					return this.setState({ embedUrl: '', embedTitle: '' });
+				}
+
+				const queryParams = `?type=${getEmbedType(input)}&input=${input}`;
+				// @ts-expect-error ts-migrate(2554) FIXME: Expected 2 arguments, but got 1.
+				return apiFetch(`/api/editor/embed${queryParams}`).then((result) => {
+					this.setState({
+						embedUrl: result.html,
+						embedTitle: result.title,
+					});
+				});
+			},
+		);
+	}
+
+	handleInsert() {
+		this.props.onInsert('iframe', {
+			url: `data:text/html;charset=utf-8,${encodeURIComponent(this.state.embedUrl)}`,
+			caption: this.state.embedTitle,
+			align: 'full',
+		});
+	}
+
+	render() {
+		return (
+			<div className="formatting-bar_media-component-content">
+				<InputGroup
+					className="top-input"
+					fill={true}
+					placeholder="Enter Twitter URL"
+					large={true}
```

Comment:
These `attr={true}` can be simplified into just `attr` just like HTML attributes.